cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/PickWinningRevision.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore;
+
+import com.cloudant.android.ContentValues;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.util.CouchUtils;
+import com.cloudant.sync.util.DatabaseUtils;
+
+import java.sql.SQLException;
+import java.util.Comparator;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Callable that evaluates leaf nodes to identify the winning revision for a particular document ID.
+ * Marks all revisions as not current, before marking a single winner.
+ *
+ * @api_private
+ */
+public class PickWinningRevision {
```

Comment:
As per my comment at https://github.com/cloudant/sync-android/pull/336#discussion_r75481031 this is not actually a `Callable`.

I preferred `mark` to `pick` for the method because it actually changes something - I'm happy to update the class to match or rename both if someone can think of something better.